src/components/CalendarPicker/generateMonthMatrix.js

```diff
@@ -0,0 +1,60 @@
+import moment from 'moment';
+
+/**
+ * Generates a matrix representation of a month's calendar given the year and month.
+ *
+ * @param {number} year - The year for which to generate the month matrix.
+ * @param {number} month - The month (0-indexed) for which to generate the month matrix.
+ * @returns {Array.<Array.<number|null>>} - A 2D array of the month's calendar days, with null values representing days outside the current month.
+ */
+export default function generateMonthMatrix(year, month) {
+    if (typeof year !== 'number') {
+        throw new TypeError('Year must be a number');
+    }
+    if (year < 0) {
+        throw new TypeError('Year must be a positive integer');
```

Comment:
Should this be an `Error` ? Also should we match the `month < 0` error message below?

```suggestion
        throw new Error('Year cannot be less than 0');
```